nominal/core/datasource_container.py

```diff
@@ -0,0 +1,328 @@
+from __future__ import annotations
+
+import abc
+import datetime
+import logging
+from typing import Iterable, Literal, Mapping, Protocol, Sequence
+
+from typing_extensions import TypeAlias, deprecated
+
+from nominal.core._clientsbunch import HasScoutParams
+from nominal.core._utils import HasRid, rid_from_instance_or_string
+from nominal.core.attachment import Attachment
+from nominal.core.connection import Connection, _iter_connections
+from nominal.core.dataset import Dataset, _get_datasets
+from nominal.core.datasource import DataSource
+from nominal.core.log import LogSet, _get_log_set
+from nominal.core.video import Video, _get_video
+
+ScopeType: TypeAlias = "Connection | Dataset | LogSet | Video"
+ScopeTypeSpecifier: TypeAlias = Literal["dataset", "video", "connection", "logset"]
+
+logger = logging.getLogger(__name__)
+
+
+class _DatasourceContainerClients(
```

Comment:
This can still be `_Clients` in the class which is then used as `DataSourceContainer._Clients` in the subclasses, no?